code-styling/markdown/README.md

```diff
@@ -0,0 +1,1155 @@
+This guide defines the Markdown coding standards we use at Dealerdirect. The intent of this guide is to reduce cognitive
+friction when scanning the Markdown source from different author, gain documentation of higher quality and gain more
+compatibility across different Markdown renders.
+
+For almost every language a style guide exists, somehow Markdown is an exception. There are several attempts,
+but none of the guides seems to be complete or they just simply cover (almost) every single option Markdown
+has to offer. As a result, the Dealerdirect Markdown style guide was born.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119][rfc2119].
+
+## Table of contents
+
+- [1. Basics](#1-basics)
+
+  - [1.1. Filenames](#11-filenames)
+  - [1.2. Files](#12-files)
+  - [1.3. Language](#13-language)
+
+- [2. Headings](#2-headings)
+
+- [3. Emphasis](#3-emphasis)
+
+- [4. Blockquotes](#4-blockquotes)
+
+- [5. Lists](#5-lists)
+
+  - [5.1. Unordered lists](#51-unordered-lists)
+  - [5.2. Ordered lists](#52-ordered-lists)
+  - [5.3. Task lists](#53-task-lists)
+
+- [6. Horizontal rules](#6-horizontal-rules)
+
+- [7. Code](#7-code)
+
+  - [7.1. Inline code](#71-inline-code)
+  - [7.2. Code blocks](#72-code-blocks)
+
+- [8. Links](#8-links)
+
+  - [8.1. Inline links](#81-inline-links)
+  - [8.2. Reference links](#82-reference-links)
+  - [8.3. Automatic links](#83-automatic-links)
+  - [8.4. Linking to books, blog and other external documents](#84-linking-to-books-blogs-and-other-external-documents)
+
+- [9. Images](#9-images)
+
+- [10. Tables](#10-tables)
+
+- [11. Emoji](#11-emoji)
+
+- [12. HTML](#12-html)
+
+## 1. Basics
+
+## 1.1. Filenames
+
+- All Markdown files MUST have the `.md` file extension.
+
+- It is RECOMMENDED to base filenames on the top-header level title.
+
+- Filenames MUST be lowercase, unless it is one of the following:
+  - `README.md`
+  - `LICENSE.md`
+  - `CONTRIBUTING.md`
+  - `CHANGELOG.md`
+
+- Filenames MUST NOT start with articles (e.g. `the`, `a`, `an`).
+
+- Filenames MUST NOT contain special characters, it is RECOMMENDED to replace them with hyphens.
+
+- Filenames MUST NOT contain spaces and/or punctuation characters, it is RECOMMENDED to replace them with hyphens.
+
+- Filenames MUST NOT start or end with a hyphen.
+
+- Filenames MUST NOT have consecutive hyphens.
+
+Good filenames:
+
+```
+markdown-style-guide.md
+introduction-into-coding.md
+hello-world.md
+```
+
+Bad filenames:
+
+```
+.markdown-style.guide.md
+-introduction-into-coding.md
+the-hello-world-example.md
+double--hypens.md
+```
+
+### 1.2 Files
+
+- Files MUST use the Unix LF (line feed) line ending.
+
+- Files MUST only use UTF-8 encoding without BOM.
+
+- Files MUST end with a single blank line.
+
+- Lines SHOULD NOT be longer than 120 characters; lines longer SHOULD be split into multiple subsequent lines.
```

Comment:
Screens are not tiny but for a side-by-side diff only half the screen is available. Even when my browser is full screen, I still can't get two lines of 120 characters in a side by side diff. 80 characters is fine, and I am not aiming for a hard limit.

Also, since this guide is public, it could be used by other companies, either directly or as a base version. With accessibility in mind (because not everybody is like you), 80 characters would make more sense. [Refreshable braille displays](https://en.wikipedia.org/wiki/Refreshable_braille_display) are available with a width of 40 or 80 characters, but not for 120 characters.